src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -38,6 +38,11 @@ pub fn collect_intra_doc_links(krate: Crate, cx: &DocContext<'_>) -> Crate {
     }
 }
 
+enum ErrorKind {
+    ResolutionFailure,
+    AnchorFailure(&'static str),
```

Comment:
I wonder if, rather than this static string, it might be nicer to have an enumeration of the anchor failure kinds and a display impl for that.  It'd better ensure that messages don't diverge if created from different parts of the code.

Reply:
I can give the type as string if you want and the rest of the message will be associated with the enum variant if you prefer? Don't really want to implement display tools on this enum.

Reply:
If you really want to avoid `impl Display` then okay :D  No need to change it for now.